### PR TITLE
python3Packages.pydruid: init at 0.6.5

### PR DESCRIPTION
**Pull request by cpcloud**
###### Description of changes

Initialize the `pydruid` package at version 0.6.5.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/pydruid/default.nix [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    requests
+  ];
+
+  nativeCheckInputs = [
+    pytestCheckHook
+    sqlalchemy
+    tornado
+    pandas
+    pycurl
+  ];
```
Comment (mweinelt) [inline pkgs/development/python-modules/pydruid/default.nix:48]:
```suggestion
  nativeCheckInputs = [
    pytestCheckHook
    pycurl
  ] ++ lib.flatten (lib.attrValues passthru.optional-dependencies);
```

At some point it also makes sense to include all optional-dependencies in `nativeCheckInputs`. Not saying this is the point, given that you currently don't pass the `cli` extra deps, but it is helpful to keep this expression in mind.

Comment (cpcloud) [reply pkgs/development/python-modules/pydruid/default.nix:48]: Roger that. I just included optional dependencies until the tests passed.

Comment (mweinelt) [reply pkgs/development/python-modules/pydruid/default.nix:48]: In which case you should also watch out for skipped tests.

Comment (cpcloud) [reply pkgs/development/python-modules/pydruid/default.nix:48]: Indeed.

In this case there are no skipped tests:

```
python3.10-pydruid> ============================= test session starts ==============================
python3.10-pydruid> platform linux -- Python 3.10.10, pytest-7.2.1, pluggy-1.0.0
python3.10-pydruid> rootdir: /build/source
python3.10-pydruid> collected 108 items
python3.10-pydruid> tests/test_async_client.py .....                                         [  4%]
python3.10-pydruid> tests/test_client.py .....                                               [  9%]
python3.10-pydruid> tests/test_query.py ........                                             [ 16%]
python3.10-pydruid> tests/db/test_cursor.py .......                                          [ 23%]
python3.10-pydruid> tests/db/test_druid_dialect.py ....                                      [ 26%]
python3.10-pydruid> tests/db/test_rows_from_chunks.py .....                                  [ 31%]
python3.10-pydruid> tests/utils/test_aggregators.py .....                                    [ 36%]
python3.10-pydruid> tests/utils/test_dimensions.py ..........................                [ 60%]
python3.10-pydruid> tests/utils/test_filters.py ............................                 [ 86%]
python3.10-pydruid> tests/utils/test_having.py .............                                 [ 98%]
python3.10-pydruid> tests/utils/test_query_utils.py ..                                       [100%]
python3.10-pydruid> ============================= 108 passed in 0.70s ==============================
```

**Comment by cpcloud** [comment]
It seems like the package produces a `pydruid` binary no matter what's installed, so perhaps it makes sense to include the `cli` dependencies by default?

**Comment by cpcloud** [comment]
It also appears that the `cli` extra is missing a `sqlalchemy` dependency since running the CLI fails with an import error without `sqlalchemy` installed.

**Comment by cpcloud** [comment]
... and finally, the version of `pygments` in nix isn't compatible with `pydruid` 😞

**Comment by cpcloud** [comment]
I suppose I can patch out the console script, and leave off the CLI dependencies, so that we don't produce a broken install.

**Comment by mweinelt** [comment]
Or expose the CLI on top-level and include the executable there, but not in the module.

**Comment by cpcloud** [comment]
Exposing it at all seems like the wrong thing to do since it doesn't work.

File: pkgs/development/python-modules/pydruid/default.nix [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+  src = fetchFromGitHub {
+    repo = pname;
+    owner = "druid-io";
+    rev = "refs/tags/${version}";
+    hash = "sha256-9+xomjSwWDVHkret/mqAZKWOPFRMvVB3CWtFPzrT81k=";
+  };
+
+  postPatch = ''
+    substituteInPlace setup.py --replace '"console_scripts": ["pydruid = pydruid.console:main"],' ""
```
Comment (mweinelt) [inline pkgs/development/python-modules/pydruid/default.nix:32]: This really should come with a comment.

Comment (cpcloud) [reply pkgs/development/python-modules/pydruid/default.nix:33]: Done.

**Comment by cpcloud** [comment]
@ofborg build python310Packages.pydruid

**Comment by cpcloud** [comment]
@ofborg build python311Packages.pydruid

**Pull request merged**